types/time.go

```diff
@@ -502,10 +502,9 @@ func (t *Time) Check() error {
 func (t *Time) Sub(t1 *Time) Duration {
 	var duration gotime.Duration
 	if t.Type == mysql.TypeTimestamp && t1.Type == mysql.TypeTimestamp {
-		// TODO: Consider time_zone variable.
-		a, err := t.Time.GoTime(gotime.Local)
+		a, err := t.Time.GoTime(t.TimeZone)
```

Comment:
Can we make sure that the `t.TimeZone` is always not nil?

Reply:
It should not be nil for timestamp... I think I can...
/run-all-tests
to check it.

Reply:
Could you add a test case for this?

Reply:
Not in this PR, because it needs to change the system timezone to reproduce it. @shenli